src/llmcompressor/modifiers/awq/base.py

```diff
@@ -382,30 +383,50 @@ def _setup_activation_cache_hooks(self) -> None:
         calculate the dynamic range during calibration
         """
 
-        def create_cache_activation_hook(smooth_layer_name):
-            def cache_activation_hook_fn(
+        def cache_parent_kwargs_hook(
+            module: torch.nn.Module,
+            args: Tuple[torch.Tensor, ...],
+            kwargs,
+        ):
+            values = inspect.signature(module.forward).bind(*args, **kwargs)
+            self._parent_kwargs_cache[module].append(values.arguments)
```

Comment:
It doesn't seem like these values are offloaded to the CPU, meaning that you're storing N x L samples on the GPU, which creates a memory bottleneck. This is where using the activations_cache offloading logic might be useful